### PR TITLE
TST: GH32822 data types in frame manip

### LINKED ISSUE #32822: dtype change when doing operations on dataframes
#### Code Sample

```python
    In [1]: import pandas as pd
    In [2]: import numpy as np
    In [3]: s = pd.Series([1,2,np.nan], dtype='Int64')
    In [4]: t = pd.Series([1,2,3], dtype='Int64')
    In [5]: s-t
    Out[5]:
    0       0
    1       0
    2    <NA>
    dtype: Int64
    In [6]: (s-t).dtype
    Out[6]: Int64Dtype()
    In [7]: s.to_frame() - t.to_frame()
    Out[7]:
          0
    0     0
    1     0
    2  <NA>
    In [8]: (s.to_frame() - t.to_frame())[0].dtype
    Out[8]: dtype('O')
```
#### Problem description

The Int64 dtype is changed when doing operations on two dataframes whereas it is kept (expected behavior) on series operations.

#### Expected Output
```
    # In [8]: (s.to_frame() - t.to_frame())[0].dtype
    # Out[8]: dtype('Int64Dtype()')
```
#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.4.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 18.7.0
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : en_GB.UTF-8

pandas           : 1.0.3
numpy            : 1.18.2
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 46.0.0
Cython           : None
pytest           : 5.4.1
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.1
IPython          : 7.13.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 0.16.0
pytables         : None
pytest           : 5.4.1
pyxlsb           : None
s3fs             : 0.4.0
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : 0.8.6
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None
</details>

### PR DESCRIPTION
**Pull request by Aadharsh-Acharya**
- [ ] closes #32822
- [ ] [1 Test added and passed]
- [ ] All [code checks passed]

### CODE REVIEW

File: pandas/tests/frame/test_arithmetic.py [hunk 1]
```diff
@@ -2040,3 +2040,13 @@ def _constructor_sliced(self):
...
     result = sdf + sdf
     tm.assert_frame_equal(result, expected)
+
+
+def test_frame_sub_dtype():
+    # GH 32822
+    s = Series([1, 2, np.nan], dtype='Int64')
+    t = Series([1, 2, 3], dtype='Int64')
+
+    expected = (s - t).dtype
```
Comment (phofl) [inline pandas/tests/frame/test_arithmetic.py:2050]: Could you please validate the whole objects with tm.assert_series_equals and please use multiletter variables

**Comment by Aadharsh-Acharya** [comment]
Does this look better?

**Comment by phofl** [comment]
Yes, could you parametrize over the nullable dtypes?

pre-commit seems to be failing

**Comment by Aadharsh-Acharya** [comment]
Absolutely, I appreciate the guidance! And I noticed the pre-commit issue, too. There seems to be a version issue on my local machine (between click and black), which isn't letting me check the pre-commit before committing the code.

File: pandas/tests/frame/test_arithmetic.py [hunk 2]
```diff
@@ -2006,6 +2006,16 @@ def test_bool_frame_mult_float():
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.parametrize("d_type", ["int8", "int16", "int32", "int64"])
```
Comment (mroeschke) [inline pandas/tests/frame/test_arithmetic.py:2009]: Could you use the `any_int_dtype` fixture instead?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @Aadharsh-Acharya

**Pull request merged**